cmd/kubectl_token.go

```diff
@@ -434,45 +442,45 @@ func samlAuth(input *LoginInput, client *http.Client) (managementClient.Token, e
 
 	tokenURL := fmt.Sprintf(authTokenURL, input.server, id)
 
-	req, err := http.NewRequest(http.MethodGet, tokenURL, bytes.NewBuffer(nil))
+	getReq, err := http.NewRequest(http.MethodGet, tokenURL, bytes.NewBuffer(nil))
 	if err != nil {
 		return token, err
 	}
+	getReq.Header.Set("content-type", "application/json")
+	getReq.Header.Set("accept", "application/json")
 
-	req.Header.Set("content-type", "application/json")
-	req.Header.Set("accept", "application/json")
-
-	client.Timeout = 300 * time.Second
+	deleteReq, err := http.NewRequest(http.MethodDelete, tokenURL, bytes.NewBuffer(nil))
+	if err != nil {
+		return token, err
+	}
+	deleteReq.Header.Set("content-type", "application/json")
+	deleteReq.Header.Set("accept", "application/json")
 
-	loginRequest := fmt.Sprintf("%s/dashboard/auth/login?requestId=%s&publicKey=%s&responseType=%s",
+	loginURL := fmt.Sprintf("%s/dashboard/auth/login?requestId=%s&publicKey=%s&responseType=%s",
 		input.server, id, encodedKey, responseType)
 
-	customPrint(fmt.Sprintf("\nLogin to Rancher Server at %s \n", loginRequest))
+	customPrint(fmt.Sprintf("\nLogin to Rancher Server at %s \n", loginURL))
 
 	interrupt := make(chan os.Signal, 1)
 	signal.Notify(interrupt, os.Interrupt)
 
-	// timeout for user to login and get token
-	timeout := time.NewTicker(15 * time.Minute)
+	// Timeout for the login flow.
+	timeout := time.NewTimer(15 * time.Minute)
 	defer timeout.Stop()
-
+	// Poll for the auth token every 10 seconds.
 	poll := time.NewTicker(10 * time.Second)
 	defer poll.Stop()
 
+loop:
 	for {
 		select {
 		case <-poll.C:
-			res, err := client.Do(req)
+			// Fetch the auth token.
+			_, respBody, err := doRequest(client, getReq)
 			if err != nil {
 				return token, err
```

Comment:
Ditto: Is it ok to return both error and token here ? Should it be the error alone instead, without the token ?